### PR TITLE
Vassal conversion code improvement & Monster Hunter changes

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Moves Bloodsucker & Vassal conversion helpers and checks into their own file instead of being thrown in with the rulesets
Reworks them to be easier to read, and fixes Vassalizing people whose masters broke the Masquerade.
Makes better var names for all the procs and documents all of them
Also removes AmInvalidAntag

### Monster Hunter changes

Monster hunters are now giving antag datums based off of a weight, with Blood cultists and Wizard added to the mix.
Monster Hunters now will also no longer spawn people who can't be owned as antags, nor will they count them towards their required list of monsters.

## Why It's Good For The Game

The code is less more resiliant to breaking and is more readable for contributors.
Also fixes a mechanic that was added a while ago, if it ever worked.

## Changelog

:cl:
fix: Deactivating a Power will no longer make you use the power when you next click.
fix: Bloodsuckers can now steal Vassals of those who broke the Masquerade.
fix: You can no longer turn invalid mobs (like Plasmamen) into special Vassals.
fix: Monster Hunters will no longer create Plasmamen Bloodsuckers and instantly kill them,
/:cl:

### CODE REVIEW

File: fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm [hunk 1]
```diff
@@ -1,3 +1,13 @@
+///List of antagonists that are considered 'Monsters' and their chance of being selected.
+GLOBAL_LIST_INIT(monster_antagonist_types, list(
+	/datum/antagonist/bloodsucker = 50,
+	/datum/antagonist/heretic = 50,
+	/datum/antagonist/changeling = 20,
+	/datum/antagonist/cult = 3,
+	/datum/antagonist/wizard/apprentice/imposter = 1,
+))
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:8]: mfw values don't add to 100

**Comment by OneAsianTortoise** [comment]
Hey man it is FFF

File: fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm (hunk 1 above)
Comment (remove32) [reply fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:8]: relative weights woah woah woah

Comment (JohnFulpWillard) [reply fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:8]: They don't have to

Comment (SgtHunk) [reply fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:8]: would be a lot cooler if they did

Comment (JohnFulpWillard) [reply fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:8]: It doesn't matter it's just weight

File: fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm [hunk 2]
```diff
@@ -42,31 +51,26 @@
...
 
 /datum/dynamic_ruleset/midround/monsterhunter/proc/generate_monsters(amount)
-	var/list/possible_monsters = list(/datum/antagonist/bloodsucker,
-	/datum/antagonist/heretic,
-	/datum/antagonist/changeling)
 	for(var/i in 1 to amount)
 		var/mob/living/monster = pick(living_players)
+		var/datum/antagonist/profession = pick_weight(GLOB.monster_antagonist_types)
+		if(!monster.mind.add_antag_datum(profession))
+			continue
```
Comment (SmoSmoSmoSmok) [inline fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:58]: this iteration will be counted in the loop, creating 1 less monster than we need everytime the monster can't be given the antag

Comment (JohnFulpWillard) [reply fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:62]: it adds to the amount to add if it fails

**Review by ghost — APPROVED** [review]

**Comment by Carlito-byte** [comment]
bad code, made it so if you get converted as a vassal you can't even see the hud mark of your master. relogging doesn't fix it either.

**Comment by JohnFulpWillard** [comment]
Bloodsucker POV
image
image

Vassal POV
image
image

**Pull request merged**